### PR TITLE
Fix issue 741 in token_info

### LINKED ISSUE #741: Deserialization bug in token_info.go
### Description

While creating unit tests for file `token_info.go` a bug was discovered inside the `_TokenInfoFromProtobuf()`.

The code snippet is checking for `AutoRenewAccount` and assigning `treasury = *_AccountIDFromProtobuf(pb.AutoRenewAccount)`

```
var treasury AccountID
  if pb.AutoRenewAccount != nil {
  treasury = *_AccountIDFromProtobuf(pb.AutoRenewAccount)
}
```

the code snipped should be refactored to:
```
var treasury AccountID
if pb.Treasury != nil {
treasury = *_AccountIDFromProtobuf(pb.Treasury)
}
```

### Steps to reproduce

func TestTokenInfo_Protobuf(t *testing.T) {
	tokenInfo := setupTokenInfo()
	pb := tokenInfo._ToProtobuf()
	expected := _TokenInfoFromProtobuf(pb)
	assert.Equal(t, tokenInfo.Treasury, expected.Treasury)
}

### Additional context

_No response_

### Hedera network

mainnet, testnet, previewnet

### Version

v2.24.2

### Operating system

None

### PR DESCRIPTION
**Pull request by Coiling-Dragon**
**Description**:

This PR fixes a bug inside token_info.go file in `TokenInfoFromProtobuf()` function

**Related issue(s)**:

Fixes #741

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: token_info_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ */
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTokenInfo_Protobuf(t *testing.T) {
```
Comment (bamzedev) [inline token_info_unit_test.go:33]: Better to be consistent with the naming convention of the project, i.e. `TestUnit....`

File: token_info_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	tokenInfo.PauseStatus = &_true
+
+	pb := tokenInfo._ToProtobuf()
+	actual := _TokenInfoFromProtobuf(pb)
+
+	assertTokenInfo(t, tokenInfo, actual)
+}
+
+func setupTokenInfo() TokenInfo {
+	publickKey, _ := PublicKeyFromStringEd25519("302a300506032b6570032100996b8090805be647ff07399e30ab1423fd3fcd795410df3c344fcd98ba27122e")
```
[the remark this answers is not in the document]
Comment (Coiling-Dragon) [reply token_info_unit_test.go:66]: Thanks, you are right.

File: token_info_unit_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		NewCustomFixedFee().
+			SetAmount(1).
+			SetDenominatingTokenID(tokenId).
+			SetFeeCollectorAccountID(feeCollectorAccountId),
+	}
+
+	return TokenInfo{
+		TokenID:             tokenId,
+		Name:                "test",
+		Symbol:              "test",
```
[the remark this answers is not in the document]
Comment (Coiling-Dragon) [reply token_info_unit_test.go:89]: Good point, will change it.

File: token_info_unit_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		Decimals:            8,
+		TotalSupply:         10000,
+		Treasury:            accId,
+		AdminKey:            publickKey,
```
[the remark this answers is not in the document]
Comment (Coiling-Dragon) [reply token_info_unit_test.go:93]: Yea, will add more keys

File: token_info_unit_test.go (hunk 1 above)
Comment (Coiling-Dragon) [reply token_info_unit_test.go:33]: Ye, my bad. Will fix it, thanks!

**Review by bamzedev — APPROVED** [review]

**Pull request merged**